lib/datadog/tracing/correlation.rb

```diff
@@ -21,26 +21,24 @@ class Identifier
         LOG_ATTR_VERSION = 'dd.version'
         LOG_ATTR_SOURCE = 'ddsource'
 
-        attr_reader \
-          :env,
-          :service,
-          :span_id,
-          :version
+        attr_reader :span_id
 
         # @!visibility private
-        def initialize(
-          env: nil,
-          service: nil,
-          span_id: nil,
-          trace_id: nil,
-          version: nil
-        )
-          # Dup and freeze strings so they aren't modified by reference.
-          @env = Core::Utils::SafeDup.frozen_dup(env || Datadog.configuration.env)
-          @service = Core::Utils::SafeDup.frozen_dup(service || Datadog.configuration.service)
+        def initialize(trace_id: nil, span_id: nil)
```

Comment:
How about keeping the constructor configurable, but not dupe configuration values to increase performance:
```ruby
@env = env ? Core::Utils::SafeDup.frozen_dup(env) : configuration.env
```

Reply:
I removed `Core::Utils::SafeDup.frozen_dup` completely